### PR TITLE
Openengsb 1269

### PR DESCRIPTION
**Pull request by michaelgis**

### CODE REVIEW

**Comment by michaelgis** [comment]
resolved issue

**Comment by michaelgis** [comment]
corrected changes

**Comment by anpieber** [comment]
again, your commits are thrown up... please correct this in a new branch

File: src/main/java/org/openengsb/connector/jira/internal/JiraService.java [hunk 1]
```diff
@@ -231,19 +231,36 @@ public class JiraService extends AbstractOpenEngSBService implements IssueDomain
...
+            rfv.setId(targetField);
+            
+            String targetValue = changes.get(attribute);
             if (targetField != null && targetValue != null) {
-                RemoteFieldValue rfv = new RemoteFieldValue();
-                rfv.setId(targetField);
-                rfv.setValues(new String[]{ targetValue });
+                if (targetField.equals("components")) {
+                    if (targetValue.contains(",")) {
+                       //Name of component must not contain ,
```
Comment (anpieber) [inline src/main/java/org/openengsb/connector/jira/internal/JiraService.java]: ok, only minor; though please move the comment ABOVE the if (you're reading code as every book; top down :)

File: src/main/java/org/openengsb/connector/jira/internal/JiraService.java [hunk 2]
```diff
@@ -231,19 +231,36 @@ public class JiraService extends AbstractOpenEngSBService implements IssueDomain
...
+                        String[] splittedComps = targetValue.split(",");
+                        String[] comps = new String[splittedComps.length];
+                        for (int i = 0; i < splittedComps.length; i++) {
+                            comps[i] = convertComponent(splittedComps[i], projComps).getId();
+                        }
+                        rfv.setValues(comps);
```
Comment (anpieber) [inline src/main/java/org/openengsb/connector/jira/internal/JiraService.java:264]: I would expect 1-2 log statements around here

File: src/test/java/org/openengsb/connector/jira/internal/JiraServiceUT.java [hunk 3]
```diff
@@ -90,8 +100,12 @@ public class JiraServiceUT {
         Issue issue = new Issue();
         issue.setSummary("summary");
         issue.setDescription("description");
-        issue.setReporter("reporter");
-        issue.setOwner("");
+        /*List<String> l = new ArrayList<String>();
```
Comment (anpieber) [inline src/test/java/org/openengsb/connector/jira/internal/JiraServiceUT.java]: please; no comments in the code!

**Comment by anpieber** [comment]
please increas the log density a little bit; otherwise this looks fine now

**Comment by michaelgis** [comment]
corrected changes

File: src/test/java/org/openengsb/connector/jira/internal/JiraServiceUT.java [hunk 4]
```diff
@@ -67,7 +70,14 @@ public class JiraServiceUT {
     public void testAddComment() {
         LOGGER.debug("test to add a command to an issue");
         jiraClient.addComment(issueId, "comment");
-
+    }
+    
+    @Test
+    public void testUpdateIssue() {
```
Comment (anpieber) [inline src/test/java/org/openengsb/connector/jira/internal/JiraServiceUT.java]: pls, all tests have to confirm to the schema: testXYZ_shouldABC

File: src/main/java/org/openengsb/connector/jira/internal/JiraService.java [hunk 5]
```diff
@@ -259,6 +288,16 @@ public class JiraService extends AbstractOpenEngSBService implements IssueDomain
         remoteIssue.setAssignee(engsbIssue.getOwner());
         remoteIssue.setProject(projectKey);
 
+        List<String> engsbComps = engsbIssue.getComponents();
```
Comment (anpieber) [inline src/main/java/org/openengsb/connector/jira/internal/JiraService.java:291]: this sounds like a private subethod (the entire added block here)

File: src/main/java/org/openengsb/connector/jira/internal/JiraService.java [hunk 6]
```diff
@@ -270,6 +309,23 @@ public class JiraService extends AbstractOpenEngSBService implements IssueDomain
 
         return remoteIssue;
     }
+    
+    private RemoteComponent convertComponent(String component, RemoteComponent[] projComps) {
+        LOGGER.info("Converting string to a component if available");
```
Comment (anpieber) [inline src/main/java/org/openengsb/connector/jira/internal/JiraService.java]: which string? Feeld free to also pack some data into your log-statements :)

File: src/main/java/org/openengsb/connector/jira/internal/JiraService.java [hunk 7]
```diff
@@ -270,6 +309,23 @@ public class JiraService extends AbstractOpenEngSBService implements IssueDomain
...
+        RemoteComponent c = new RemoteComponent();
+        try {
+            Integer.parseInt(component);
+            c.setId(component);   
+        } catch (NumberFormatException e) {
+            for (RemoteComponent tmpComp : projComps) {
```
Comment (anpieber) [inline src/main/java/org/openengsb/connector/jira/internal/JiraService.java:328]: I think you can add one log here too

File: src/main/java/org/openengsb/connector/jira/internal/JiraService.java [hunk 8]
```diff
@@ -259,6 +288,16 @@ public class JiraService extends AbstractOpenEngSBService implements IssueDomain
         remoteIssue.setAssignee(engsbIssue.getOwner());
         remoteIssue.setProject(projectKey);
 
+        List<String> engsbComps = engsbIssue.getComponents();
+        if (engsbComps != null) {
```
Comment (anpieber) [inline src/main/java/org/openengsb/connector/jira/internal/JiraService.java:292]: here's missing a log somewhere

File: src/main/java/org/openengsb/connector/jira/internal/JiraService.java [hunk 9]
```diff
@@ -251,7 +279,8 @@ public class JiraService extends AbstractOpenEngSBService implements IssueDomain
         return remoteFieldArray;
     }
 
-    private RemoteIssue convertIssue(Issue engsbIssue) {
+    private RemoteIssue convertIssue(Issue engsbIssue, JiraSoapService jiraSoapService) throws RemoteException {
+        LOGGER.info("Converting openengsb issue to remote issue");
```
Comment (anpieber) [inline src/main/java/org/openengsb/connector/jira/internal/JiraService.java]: which issue?

File: src/main/java/org/openengsb/connector/jira/internal/JiraService.java [hunk 10]
```diff
@@ -231,19 +239,39 @@ public class JiraService extends AbstractOpenEngSBService implements IssueDomain
...
+            
+            String targetValue = changes.get(attribute);
             if (targetField != null && targetValue != null) {
-                RemoteFieldValue rfv = new RemoteFieldValue();
-                rfv.setId(targetField);
-                rfv.setValues(new String[]{ targetValue });
+                if (targetField.equals("components")) {
+                  //Name of component must not contain ,
+                    if (targetValue.contains(",")) {
+                        LOGGER.info("adding more than one component");
```
Comment (anpieber) [inline src/main/java/org/openengsb/connector/jira/internal/JiraService.java:257]: somewhere you may like to add: which ones

File: src/main/java/org/openengsb/connector/jira/internal/JiraService.java [hunk 11]
```diff
@@ -192,6 +197,8 @@ public class JiraService extends AbstractOpenEngSBService implements IssueDomain
                 report.add("\n");
             }
 
+            LOGGER.info("Successfully created release report " + releaseId);
```
Comment (anpieber) [inline src/main/java/org/openengsb/connector/jira/internal/JiraService.java]: SLF4J comes with a MUCH better system für appending strings... please take a look at the slf4j docs or at some other places at the openengsb where we use them too

File: src/main/java/org/openengsb/connector/jira/internal/JiraService.java [hunk 12]
```diff
@@ -131,6 +133,8 @@ public class JiraService extends AbstractOpenEngSBService implements IssueDomain
             for (RemoteIssue issue : issues) {
                 jiraSoapService.updateIssue(authToken, issue.getKey(), changes);
             }
+            
+            LOGGER.info("Successfully moved " + releaseFromId + " to " + releaseToId);
```
Comment (anpieber) [inline src/main/java/org/openengsb/connector/jira/internal/JiraService.java]: SLF4J comes with a MUCH better system für appending strings... please take a look at the slf4j docs or at some other places at the openengsb where we use them too

File: src/main/java/org/openengsb/connector/jira/internal/JiraService.java [hunk 13]
```diff
@@ -156,6 +160,7 @@ public class JiraService extends AbstractOpenEngSBService implements IssueDomain
                 return;
             }
             jiraSoapService.releaseVersion(authToken, projectKey, version);
+            LOGGER.info("Successfully closed release " + id);
```
Comment (anpieber) [inline src/main/java/org/openengsb/connector/jira/internal/JiraService.java]: SLF4J comes with a MUCH better system für appending strings... please take a look at the slf4j docs or at some other places at the openengsb where we use them too

File: src/main/java/org/openengsb/connector/jira/internal/JiraService.java [hunk 14]
```diff
@@ -102,8 +103,9 @@ public class JiraService extends AbstractOpenEngSBService implements IssueDomain
         try {
             JiraSoapService jiraSoapService = login();
 
-            RemoteFieldValue[] values = convertChanges(changes);
+            RemoteFieldValue[] values = convertChanges(changes, jiraSoapService);
             jiraSoapService.updateIssue(authToken, issueKey, values);
+            LOGGER.info("Successfully updated issue {}", issueKey);
```
Comment (anpieber) [inline src/main/java/org/openengsb/connector/jira/internal/JiraService.java:108]: yeah, exactly this one :)

File: src/main/java/org/openengsb/connector/jira/internal/JiraService.java [hunk 15]
```diff
@@ -88,6 +88,7 @@ public class JiraService extends AbstractOpenEngSBService implements IssueDomain
             RemoteComment comment = new RemoteComment();
             comment.setBody(commentString);
             jiraSoapService.addComment(authToken, issueKey, comment);
+            LOGGER.info("Successfully added comment {}", commentString);
         } catch (RemoteException e) {
             LOGGER.error("Error commenting issue . XMLRPC call failed. ");
```
Comment (anpieber) [inline src/main/java/org/openengsb/connector/jira/internal/JiraService.java]: you can add "e" here

**Comment by anpieber** [comment]
almost there :)

**Comment by michaelgis** [comment]
corrected comments

**Pull request merged**